cores/arduino/UARTClass.cpp

```diff
@@ -168,7 +168,7 @@ int UARTClass::read( void )
 
 void UARTClass::flush( void )
 {
-  while (_tx_buffer->_iHead != _tx_buffer->_iTail); //wait for transmit data to be sent
+  while (_tx_buffer->_iHead != *(volatile int*)&(_tx_buffer->_iTail)); //wait for transmit data to be sent
```

Comment:
Is there a particular reason for casting the pointer to a volatile pointer first prior to using it?  Is the code being optimized out by the compiler if the volatile casting is omitted?